website/src/pages/product-ecosystem.tsx

```diff
@@ -139,8 +310,83 @@ function Index() {
 
         <CTA />
       </div>
+
+      <Modal
+        open={modalVisible}
+        onClose={onCloseModal}
+        center
+        blockScroll={false}
+        classNames={{
+          modal: `${styles.modalDialog} ${modalDialogAnimation}`,
+          overlay: `${styles.modalOverlay} ${modalOverlayAnimation}`,
+          modalAnimationIn: styles.modalSlideIn,
+        }}
+      >
+        <ModalContent />
+      </Modal>
+
+      <ReactTooltip
+        id="rightTip"
+        place="right"
+        type="dark"
+        effect="solid"
+        delayShow={0}
+        html
+        getContent={html => html}
+        className={styles.tooltipStyling}
+        event="mouseenter"
+        eventOff="mouseleave focusout"
+      />
+      <ReactTooltip
+        id="leftTip"
+        place="left"
+        type="dark"
+        effect="solid"
+        delayShow={0}
+        html
+        getContent={html => html}
+        className={styles.tooltipStyling}
+        event="mouseenter"
+        eventOff="mouseleave focusout"
+      />
     </Layout>
   )
 }
 
 export default Index
+
+function RoadmapColumn({ icon, title, className, items, clickFn }) {
+  return (
+    <div className={styles.roadmapColumn}>
+      <div className={styles.roadmapColumnTitle}>
```

Comment:
Looking at the page I think this is semantically a `h3`.
```suggestion
      <h3 className={styles.roadmapColumnTitle}>
```

But I see we're not really doing much semantic heading elsewhere in this page, so maybe just leave it for later.

Reply:
What I was going to say was that headings can break and go over two lines, breaking the "gridiness" of the roadmap

![image](https://user-images.githubusercontent.com/1413275/209106058-d040eadd-ba86-446d-a6b7-0a5edbe442d5.png)

I think the easiest fix is to just force it to be one line
```
white-space: nowrap;
overflow: hidden;
text-overflow: ellipsis;
```

With this css applied it looks like this:
![image](https://user-images.githubusercontent.com/1413275/209106330-db2e1bb6-219d-49bc-91a6-ebbecd74d1b0.png)

But maybe only for desktop/non-mobile.

The other alternative would be to make the entire roadmap _one_ grid and place the headings and elements in their correct column. Unsure how much work this would be… probably not worth it for this iteration.

Reply:
@Prid13: Had a slack discussion with @torotimes about ☝️ and think we want to go this route. I.e. no scrollbars, and everything in one grid (headers and cards), to avoid any alignment issues.

So kinda like this:
https://user-images.githubusercontent.com/1413275/209145077-e5fbe379-b600-41ce-9d67-f1989764a076.mov

And we fully realise it's very tall/uneven right now, which we will use as motivation to move items to the right in our board 😅.

I'm slightly unsure about what to do about card-content potentially spanning more than 1 line… maybe cap at 1 line with ellipsis?

Does it sound reasonable or do you foresee any issues with doing it this way? (i was thinking we could possibly merge all the card arrays together in one flat array while annotating each object with a class to set the correct grid column on them. And use `order` to get each column in the right place on mobile)

Reply:
Great feedback! 😇 I'm one to never back down from a challenge, so while this proved a bit more challenging to accomplish than anticipated, I think I've landed on a solution that works great on desktop + mobile. Not sure about the SEO aspect of it, but hopefully that's not too big of an issue 😅

I changed the grid from a 4x1 to a 4x2 container, with the titles occupying the top row, while the columns the bottom row, ensuring the top row (with titles) occupies equal height. Then I used `grid-template-areas` to strategically place the items in correct order for tablet/mobile view. For the roadmap items, I used your great idea of showing en ellipsis for longer texts, which works amazingly well in tandem with the tooltip :)

Desktop (long dummy title):
> ![image](https://user-images.githubusercontent.com/27897747/209191514-bf670873-5d5c-4eb9-ab5f-b327e415bb16.png)

Tablet/Wide Mobile:
> ![image](https://user-images.githubusercontent.com/27897747/209195445-2147b427-e955-4ba6-a881-eebad2426d5b.png)

Thanks again for taking the time to review this 😇⭐